src/test/java/org/opensearch/neuralsearch/query/HybridQueryPostFilterIT.java

```diff
@@ -0,0 +1,418 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.query;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import lombok.SneakyThrows;
+import org.junit.Before;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.index.query.MatchNoneQueryBuilder;
+import org.opensearch.index.query.TermQueryBuilder;
+import org.opensearch.index.query.RangeQueryBuilder;
+import org.opensearch.index.query.MatchQueryBuilder;
+import org.opensearch.neuralsearch.BaseNeuralSearchIT;
+import static org.opensearch.neuralsearch.util.AggregationsTestUtils.getAggregations;
+import static org.opensearch.neuralsearch.util.AggregationsTestUtils.getNestedHits;
+import static org.opensearch.neuralsearch.util.AggregationsTestUtils.getAggregationValues;
+import org.opensearch.search.aggregations.AggregationBuilder;
+import org.opensearch.search.aggregations.AggregationBuilders;
+import static org.opensearch.neuralsearch.util.TestUtils.assertHitResultsFromQuery;
+
+public class HybridQueryPostFilterIT extends BaseNeuralSearchIT {
+    private static final String TEST_MULTI_DOC_INDEX_WITH_TEXT_AND_INT_MULTIPLE_SHARDS =
+        "test-hybrid-post-filter-multi-doc-index-multiple-shards";
+    private static final String TEST_MULTI_DOC_INDEX_WITH_TEXT_AND_INT_SINGLE_SHARD =
+        "test-hybrid-post-filter-multi-doc-index-single-shard";
+    private static final String SEARCH_PIPELINE = "phase-results-hybrid-post-filter-pipeline";
+    private static final String INTEGER_FIELD_1 = "stock";
+    private static final String TEXT_FIELD_1 = "name";
+    private static final String KEYWORD_FIELD_2 = "category";
+    private static final String TEXT_FIELD_NAME_1_VALUE = "Dunes part 2";
+    private static final String TEXT_FIELD_NAME_2_VALUE = "Dunes part 1";
+    private static final String TEXT_FIELD_NAME_3_VALUE = "Mission Impossible 1";
+    private static final String TEXT_FIELD_NAME_4_VALUE = "Mission Impossible 2";
+    private static final String TEXT_FIELD_NAME_5_VALUE = "The Terminal";
+    private static final String TEXT_FIELD_NAME_6_VALUE = "Avengers";
+    private static final int INTEGER_FIELD_STOCK_1_VALUE = 25;
+    private static final int INTEGER_FIELD_STOCK_2_VALUE = 22;
+    private static final int INTEGER_FIELD_STOCK_3_VALUE = 256;
+    private static final int INTEGER_FIELD_STOCK_4_VALUE = 25;
+    private static final int INTEGER_FIELD_STOCK_5_VALUE = 20;
+    private static final String KEYWORD_FIELD_CATEGORY_1_VALUE = "Drama";
+    private static final String KEYWORD_FIELD_CATEGORY_2_VALUE = "Action";
+    private static final String KEYWORD_FIELD_CATEGORY_3_VALUE = "Sci-fi";
+    private static final String AVG_AGGREGATION_NAME = "avg_stock_size";
+    private static boolean setUpIsDone = false;
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        if (setUpIsDone) {
```

Comment:
why this is needed? Seems that the only thing we're skipping is index settings update

Reply:
That means you effectively making method annotated with `before` run as method annotated with `beforeClass`. Can you check if we can use [BeforeClass](https://junit.org/junit4/javadoc/4.12/org/junit/BeforeClass.html) directly instead reinveting the wheel

Reply:
I did that at the first place It threw and error for some instantiation

Reply:
understood, let's stick to your approach then. Just put comment with short explanation that those lines are executed in "before class" manner and this is a workaround because annotation cannot be used as it throws errors in scenario like <your_scenario_and_error>